src/legacy/ui/ui_settings/create_or_upgrade_saved_config/create_or_upgrade_saved_config.ts

```diff
@@ -18,37 +18,38 @@
  */
 
 import { defaults } from 'lodash';
+import { SavedObjectsClientContract, SavedObjectAttribute } from 'src/core/server';
+import { Legacy } from 'kibana';
 
 import { getUpgradeableConfig } from './get_upgradeable_config';
 
-export async function createOrUpgradeSavedConfig(options) {
-  const {
-    savedObjectsClient,
-    version,
-    buildNum,
-    logWithMetadata,
-    onWriteError,
-  } = options;
+interface Options {
+  savedObjectsClient: SavedObjectsClientContract;
+  version: string;
+  buildNum: number;
+  logWithMetadata: Legacy.Server['logWithMetadata'];
+  onWriteError?: <T extends SavedObjectAttribute = any>(
+    error: Error,
+    attributes: Record<string, any>
+  ) => Record<string, T> | undefined;
+}
+export async function createOrUpgradeSavedConfig<T extends SavedObjectAttribute = any>(
+  options: Options
+): Promise<Record<string, T> | undefined> {
+  const { savedObjectsClient, version, buildNum, logWithMetadata, onWriteError } = options;
```

Comment:
Move object destructure to function arguments?

Reply:
I'd prefer not to do. there are already too many params to use positional arguments.